### PR TITLE
[Node.js] Confusion about how to prune devDependencies

### PR DESCRIPTION
**Pull request by TheoPouletSouchard**
See https://github.com/Scalingo/documentation/issues/2839

### CODE REVIEW

**Comment by TheoPouletSouchard** [comment]
@EtienneM I tried something with a table, it might be easier to read, WDYT ? (i can rollback to old way if you think it was better before)

**Review by EtienneM — CHANGES_REQUESTED** [review]
I like the idea of a table, but the CLI column is too much. It does not render well:

image

Maybe remove this column, add a column "Value" and add a line after the table like:

```
scalingo --app my-app env-set variable=value
```
or something like that?

**Comment by TheoPouletSouchard** [comment]
> I like the idea of a table, but the CLI column is too much. It does not render well:
> 
> image
> 
> Maybe remove this column, add a column "Value" and add a line after the table like:
> 
> ```
> scalingo --app my-app env-set variable=value
> ```
> 
> or something like that?

Uhm i see what you mean. I'm wondering: as the values just need to be set and can be any string to work, the "value" column is useless, isn't it ? 

It might be better to:
- Remove the CLI column-
- Add the CLI line under the table (as you suggested) 
- Add a small text to say "the values can be anything as long as they are set to be valid" (written in a better way)

WDYT?

**Comment by Frzk** [comment]
Hello @TheoPouletSouchard

Thanks a lot for this PR <3

> Uhm i see what you mean. I'm wondering: as the values just need to be set and can be any string to work, the "value" column is useless, isn't it ?

- For `NPM_CONFIG_PRODUCTION`, it seems the value MUST be either `true` or `false` (See https://github.com/Scalingo/nodejs-buildpack/blob/a2a95190bcc7482f61d3dc7b4cc853d25785a117/lib/environment.sh#L61)
- For `YARN_PRODUCTION`, it seems it MUST be set to `true` to be effective. (see https://github.com/Scalingo/nodejs-buildpack/blob/a2a95190bcc7482f61d3dc7b4cc853d25785a117/lib/failure.sh#L371)
- For `YARN2_SKIP_PRUNING`, it seems it MUST be set to `true` to be effective. (see https://github.com/Scalingo/nodejs-buildpack/blob/a2a95190bcc7482f61d3dc7b4cc853d25785a117/lib/dependencies.sh#L181)

While I strongly agree that the presence/absence of the environment variable should be enough, it seems it's not the case.

Consequently, I suggest to replace the CLI column of the table with a "Value" column and put an example below.

File: src/_posts/languages/nodejs/2000-01-01-start.md [hunk 1]
```diff
@@ -98,6 +98,21 @@ By default, dependencies present in the `devDependencies` field are installed. A
 
 The pruning step can be skipped by setting the environment variable `YARN_PRODUCTION` to any value.
 
+#### Skip Pruning
+
+If you need access to packages declared under devDependencies at runtime or in a different buildpack then, depending on the package manager used, you can set one of the following environment variables to skip the pruning step:
+
+| Package manager | Version | Environment variable    | Value              |
+|-----------------|---------|-------------------------|--------------------|
+| npm             | Any     | `NPM_CONFIG_PRODUCTION` | `true`  or `false` |
```
Comment (EtienneM) [inline src/_posts/languages/nodejs/2000-01-01-start.md:107]: issue: well, from the code sent by François, the value to skip the pruning should be `true`, shouldn't it?

```suggestion
| npm             | Any     | `NPM_CONFIG_PRODUCTION` | `true`  |
```

**Review by EtienneM — CHANGES_REQUESTED** [review]

**Review by EtienneM — APPROVED** [review]
Thanks a lot!! :rocket:

**Pull request merged**